bpf/bpf_lxc.c

```diff
@@ -578,7 +578,7 @@ static __always_inline int handle_ipv4_from_lxc(struct __ctx_buff *ctx,
 	}
 
 skip_service_lookup:
-#endif /* !ENABLE_HOST_SERVICES_FULL */
+#endif /* !ENABLE_HOST_SERVICES_FULL || ENABLE_SOCKET_LB_HOST_ONLY */
```

Comment:
Don't we also need to extend the following code?https://github.com/cilium/cilium/blob/4145278ccc6e90739aa100c9ea8990a0f561ca95/bpf/bpf_lxc.c#L1373-L1382

<br>

Could we then simplify this with a new macro `ENABLE_PER_PACKET_LB = !ENABLE_HOST_SERVICES_FULL || ENABLE_SOCKET_LB_HOST_ONLY`? It would simplify the logic here and, if we want to see when `ENABLE_PER_PACKET_LB` is true, there would be a single place to look at, in userspace.

Reply:
Ouch, good catch. Also, good suggestion re the new macro.